acquia/mc-cs-plugin-custom-objects
# "in the last" and "in the next" segment filters.

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | No
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

User should able to select the days/months/years unit in filter value with interval.
- ”In the last” works as between current date minus those days/months/year and current date.
- ”In the next” works as between current date plus those days/months/year and current date.

<img width="1109" alt="image" src="https://github.com/acquia/mc-cs-plugin-custom-objects/assets/68939488/eb451b07-5195-41c9-b0f7-bea1de8385d3">


#### Steps to test this PR:


1. Open this PR for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Test "In the last" and "In the next" operator will be available for Date and DateTime type fields.
3. Test segment build with "In the last" and "In the next" operators.
